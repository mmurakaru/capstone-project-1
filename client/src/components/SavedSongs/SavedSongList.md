List of saved songs:

```js
<SavedSongList
  deleteSavedSong={() => {}}
  savedSongs={[
    {
      artist: 'Unprocessed',
      songTitle: 'Real',
      songSnippet: 'https://test/',
      id: 'a',
    },
    {
      artist: 'Dance Gavin Dance',
      songTitle: 'We Own The Night',
      songSnippet: 'https://test/',
      id: 'b',
    },
  ]}
  toggleCurrentSongId={() => {}}
  isSongPlaying={false}
  currentSongId="a"
/>
```
