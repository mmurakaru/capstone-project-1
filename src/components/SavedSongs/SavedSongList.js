import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import SavedSong from './SavedSong'

SavedSongList.propTypes = {
  stopPlayingSong: PropTypes.func,
  deleteSavedSong: PropTypes.func.isRequired,
  savedSongs: PropTypes.array.isRequired,
  toggleCurrentSongId: PropTypes.func.isRequired,
  isSongPlaying: PropTypes.bool.isRequired,
  currentSongId: PropTypes.string,
}

export default function SavedSongList({
  stopPlayingSong,
  deleteSavedSong,
  savedSongs,
  toggleCurrentSongId,
  isSongPlaying,
  currentSongId,
}) {
  return (
    <List>
      {savedSongs?.map(({ artist, songTitle, id }) => (
        <SavedSong
          key={id}
          artist={artist}
          songTitle={songTitle}
          id={id}
          stopPlayingSong={stopPlayingSong}
          deleteSavedSong={deleteSavedSong}
          toggleCurrentSongId={toggleCurrentSongId}
          isSongPlaying={isSongPlaying}
          currentSongId={currentSongId}
        />
      ))}
    </List>
  )
}

const List = styled.ul`
  display: grid;
  gap: 32px;
  list-style: none;
  margin: 22px 0;
  padding: 0;
`
