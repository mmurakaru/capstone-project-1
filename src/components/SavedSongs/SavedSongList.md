One saved song item:

```js
<SavedSongList
  savedSongs={[
    {
      artist: 'Unprocessed',
      songTitle: 'Real',
      songSnippet: 'https://test/',
      id: 'a',
    },
  ]}
  toggleCurrentSongId={() => {}}
  isSongPlaying={false}
  currentSongId="a"
/>
```
