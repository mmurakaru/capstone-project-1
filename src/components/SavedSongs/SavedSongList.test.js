import { render } from '@testing-library/react'
import SavedSongList from './SavedSongList'

describe('SavedSongList', () => {
  it('renders correctly', () => {
    const { container } = render(
      <SavedSongList
        stopPlayingSongById={() => {}}
        deleteSavedSong={() => {}}
        savedSongs={[
          {
            artist: 'Unprocessed',
            songTitle: 'Real',
            songSnippet: 'https://test/',
            id: 'a',
          },
          {
            artist: 'Dance Gavin Dance',
            songTitle: 'We Own The Night',
            songSnippet: 'https://test/',
            id: 'b',
          },
        ]}
        toggleCurrentSongId={() => {}}
        isSongPlaying={false}
        currentSongId="a"
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
