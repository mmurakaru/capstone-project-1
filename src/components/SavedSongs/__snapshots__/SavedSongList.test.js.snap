// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedSongList renders correctly 1`] = `
.c5 {
  background-color: transparent;
  border: var(--primary-dark) 2px solid;
  fill: var(--primary-dark);
  height: 46px;
  padding: 10px;
  width: 46px;
}

.c0 {
  display: grid;
  gap: 8px;
  list-style: none;
  margin: 0;
  padding: 0;
}

.c1 {
  background-color: var(--primary-light);
  display: grid;
  grid-template-columns: 80% auto;
  grid-template-rows: 1fr 1fr;
  padding: 20px;
}

.c2 {
  display: block;
  font-size: 75%;
  font-weight: 300;
  padding-bottom: 6px;
  text-transform: uppercase;
}

.c3 {
  color: var(--primary-dark);
  font-size: 1em;
  font-weight: 600;
  grid-column-start: 1;
}

.c4 {
  grid-column-start: 2;
  grid-row: 1/3;
  justify-self: end;
}

<ul
  class="c0"
>
  <li
    class="c1"
  >
    <span
      class="c2"
    >
      Unprocessed
    </span>
    <span
      class="c3"
    >
      Real
    </span>
    <div
      class="c4"
    >
      <button
        class="c5"
        data-testid="audio-button"
      >
        <svg
          title="play"
        >
          play.svg
        </svg>
      </button>
    </div>
  </li>
</ul>
`;
